azurerm/internal/services/apimanagement/resource_arm_api_management_subscription.go

```diff
@@ -43,8 +43,10 @@ func resourceArmApiManagementSubscription() *schema.Resource {
 				ValidateFunc: validation.Any(validation.IsUUID, validation.StringIsEmpty),
 			},
 
+			// 3.0 this seems to have been renamed to owner id?
 			"user_id": azure.SchemaApiManagementChildID(),
 
+			// 3.0 this seems to have been renamed to scope?
 			"product_id": azure.SchemaApiManagementChildID(),
```

Comment:
This is not a simple rename.

An apim subscription can be tied to:
- a product -> "/products/{productId}"
- all apis -> "/apis"
- a single api -> "/apis/{apiId}"

So i think you must create a new block called "scope" with all the 3 options and in the docs you must document that only one must be filled link:

```
scope {
all_apis: boolean
api_name: string
product_name: string
}
```

used as:

scope {
product_name: "myproduct"
}

and than set the scope in SubscriptionCreateParameterProperties like
Scope: "/products/myproduct"

Hope it make sense.

Reply:
thinking about this, this probably wants two fields here: `api_id` and `product_id` - both of which are optional & conflict with each other; which'll allow all 3 conditions to be met

Reply:
@francescopersico - i believe as i have coded it the original behaviour will be preserved? i'll leave adding the additional properties/block to a future PR by someone who knows this service better then I